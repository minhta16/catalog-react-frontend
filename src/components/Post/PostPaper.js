import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Divider, Paper, Button } from '@material-ui/core';
import { Link } from 'react-router-dom';

export function PostPaper(props) {
  const { header, description, price, categoryId, categoryName, created } = props;
  return (
    <Paper className="left-margin">
      <Button color="primary" component={Link} to={`/category/${categoryId}`}>
        Back
      </Button>
      <Typography variant="h4">{header}</Typography>
      <Typography variant="subtitle1">{`Created: ${created}`}</Typography>
      <Typography variant="subtitle1">{`Category: ${categoryName}`}</Typography>
      <Divider />
      <Typography variant="body1">{description}</Typography>
      <Typography variant="body1">{`$ ${price / 100}`}</Typography>
    </Paper>
  );
}

PostPaper.propTypes = {
  header: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  categoryId: PropTypes.string.isRequired,
  categoryName: PropTypes.string.isRequired,
  created: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
};

export default PostPaper;
