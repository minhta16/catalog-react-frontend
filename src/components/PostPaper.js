import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Divider, Paper, Button } from '@material-ui/core';
import { Link } from 'react-router-dom';

export function PostPaper(props) {
  const { header, body, categoryId } = props;
  return (
    <Paper className="left-margin">
      <Button color="primary" component={Link} to={`/${categoryId}`}>
        Back
      </Button>
      <Typography variant="h4">{header}</Typography>
      <Typography variant="subtitle1">By: Minh Ta</Typography>
      <Divider />
      <Typography variant="body1">{body}</Typography>
    </Paper>
  );
}

PostPaper.propTypes = {
  header: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  categoryId: PropTypes.string.isRequired,
};

export default PostPaper;
