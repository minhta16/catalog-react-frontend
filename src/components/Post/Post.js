import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PostPaper from 'components/Post/PostPaper';
import { Container } from '@material-ui/core';
import { selectPost } from 'selectors/posts';
import { selectCategory } from 'selectors/categories';

export function Post(props) {
  const { match, currentPost, category } = props;
  return (
    <Container maxWidth="lg">
      <PostPaper
        header={currentPost.name}
        created={currentPost.created}
        body={currentPost.description}
        categoryName={category.name}
        categoryId={match.params.id}
      />
    </Container>
  );
}

Post.propTypes = {
  match: PropTypes.object.isRequired,
  currentPost: PropTypes.object.isRequired,
  category: PropTypes.object.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  currentPost: selectPost(state, ownProps.match.params.postId),
  category: selectCategory(state, ownProps.match.params.id),
});
Post.propTypes = {};

export default connect(mapStateToProps)(Post);
