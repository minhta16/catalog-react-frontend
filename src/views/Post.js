import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PostPaper from 'components/PostPaper';
import { Container } from '@material-ui/core';

export function Post(props) {
  const { match, selectedCatItems } = props;
  const currentPost = selectedCatItems[match.params.postId];
  return (
    <Container maxWidth="lg">
      <PostPaper
        header={currentPost.name}
        created={currentPost.created}
        body={currentPost.description}
        categoryId={match.params.id}
      />
    </Container>
  );
}

Post.propTypes = {
  match: PropTypes.shape.isRequired,
  selectedCatItems: PropTypes.shape.isRequired,
};

const mapStateToProps = (state) => ({
  selectedCatItems: state.posts,
});
Post.propTypes = {};

export default connect(mapStateToProps)(Post);
