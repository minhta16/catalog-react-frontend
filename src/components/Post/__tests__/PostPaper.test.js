/* eslint-disable no-undef */
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import { PostPaper } from 'components/Post/PostPaper';

configure({ adapter: new Adapter() });

describe('components/PostPaper', () => {
  let wrapper;
  let props;
  // let warning;

  // const update = () => {
  //   wrapper.update();
  //   warning = wrapper.find(Warning);
  // };

  const setup = () => {
    wrapper = shallow(<PostPaper {...props} />);
  };

  beforeEach(() => {
    props = {
      header: 'header',
      description: 'body',
      categoryId: '1',
      categoryName: 'name',
      created: '2019',
      price: 100,
    };
  });

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
});
