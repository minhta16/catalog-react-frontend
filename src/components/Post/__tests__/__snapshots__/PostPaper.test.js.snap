// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PostPaper should render correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="left-margin"
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    component={[Function]}
    to="/1"
  >
    Back
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Typography))
    variant="h4"
  >
    header
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    variant="subtitle1"
  >
    Created: undefined
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    variant="subtitle1"
  >
    Category: undefined
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(Typography))
    variant="body1"
  >
    body
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Paper))>
`;
